src/mgba_printf/mini_printf.c

```diff
@@ -167,13 +274,13 @@ s32 mini_vsnprintf(char *buffer, u32 buffer_len, const char *fmt, va_list va)
     b.buffer_len = buffer_len;
     if (buffer_len == 0)
         buffer = NULL;
-    n = mini_vpprintf(_puts, (buffer != NULL) ? &b : NULL, fmt, va);
+    n = mini_vpprintf(_puts,_puts2, (buffer != NULL) ? &b : NULL, fmt, va);
     if (buffer == NULL)
         return n;
     return b.pbuffer - b.buffer;
 }
 
-s32 mini_vpprintf(s32 (*puts)(char* s, s32 len, void* buf), void* buf, const char *fmt, va_list va)
+s32 mini_vpprintf(s32 (*puts)(char* s, s32 len, void* buf),s32 (*puts2)(char* s, s32 len, void* buf), void* buf, const char *fmt, va_list va)
```

Comment:
The function essentially conforms to libc `vpprintf` which does not have a second function pointer argument. I think this should be reflected here, the alternate puts can just be called without a function pointer, essentially it is never called with another argument anyways.